tests/unit_test/linux/CMakeLists.txt

```diff
@@ -46,8 +46,8 @@
                 )
     set_target_properties(utils PROPERTIES
             LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
-            COMPILE_FLAGS "-Wall -Og -pthread"
-            LINK_FLAGS "-Wall -Og -pthread -lpthread"
+            COMPILE_FLAGS "-Wall -Og -pthread -ggdb3"
```

Comment:
Why are we enabling debugging (`-ggdb3`) without using `CMAKE_CFLAGS_DEBUG`?  Ideally we should be executing unit tests both with and without debug flags and using [`CMAKE_BUILD_TYPE`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_TYPE.html) to control the build type.

Reply:
when would you not want debugging information with the UT?

Reply:
Ideally you will run your tests with and without debugging.  But the main issue here is that we're adding debugging support in a non-cmake standard way.  cmake gives you a standard method of doing this with `CMAKE_CFLAGS_DEBUG` and `CMAKE_BUILD_TYPE`, but we're not using that here.

Reply:
Ah I see what you mean, good point, pushed another commit, thanks!

Reply:
nice!